### PR TITLE
Fix the panic after init message `prepare` failure

### PR DESCRIPTION
**Pull request by breathx**
Tests `non_existent_code_id_zero_gas` and `waited_with_zero_gas` are panicking now. Others are just for confidence.

### CODE REVIEW

File: pallets/gear/src/manager/journal.rs [hunk 1]
```diff
@@ -140,10 +143,22 @@ where
                 // dequeued. The other case is async init.
                 wake_waiting_init_msgs(program_id);
 
-                self.clean_reservation_tasks(program_id);
+                // If we run into `InitFailure` after real execution (not
+                // prepare or precharge) processor methods, then we are
```
Comment (gshep) [inline pallets/gear/src/manager/journal.rs:147]:
```suggestion
                // prepare or precharge processor methods), then we are
```

File: pallets/gear/src/manager/mod.rs [hunk 2]
```diff
@@ -268,11 +268,20 @@ where
...
-        let active_program =
-            common::get_active_program(program_id.into_origin()).unwrap_or_else(|e| {
-                unreachable!("`exit` can be called only from active program: {}", e)
-            });
+    fn clean_reservation_tasks(&mut self, program_id: ProgramId, maybe_inactive: bool) {
+        let maybe_active_program = common::get_active_program(program_id.into_origin());
+
+        if maybe_active_program.is_err() && maybe_inactive {
+            return;
+        };
```
Comment (gshep) [inline pallets/gear/src/manager/mod.rs:276]: unnecessary semicolon

**Review by gshep — APPROVED** [review]
Good catch! :gear:

File: pallets/gear/src/manager/journal.rs (hunk 1 above)
Comment (breathx) [reply pallets/gear/src/manager/journal.rs:147]: Will be applied next pr just to avoid waiting one more ci

**Pull request merged**